File: solutions/cpp/raindrops/2/raindrops.cpp
#include <array>

#include "raindrops.h"

namespace raindrops {
	const std::array<std::pair<int, std::string>, 3> drops = { {
		{3, "Pling"},
		{5, "Plang"},
		{7, "Plong"}
	} };

	std::string convert(int num) {
		std::string result = "";

		for (auto &[factor, drop] : drops)
			if (num % factor == 0)
				result += drop;

		if (result == "")
			return std::to_string(num);
		return result;
	}
}  // namespace raindrops

File: solutions/ruby/raindrops/1/raindrops.rb
=begin
Write your code for the 'Raindrops' exercise in this file. Make the tests in
`raindrops_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/raindrops` directory.
=end

class Raindrops
  def self.convert(num)
    res = ""

    if num % 3 == 0
      res << "Pling"
    end

    if num % 5 == 0
      res << "Plang"
    end

    if num % 7 == 0
      res << "Plong"
    end

    if res == ""
      res << num.to_s
    end
    
    return res
  end
end